doc/maintainers/team_process.md

```diff
@@ -178,8 +178,72 @@ maintainers team makes no guarantees about regularly checking them.
 now the most important thing is to direct users to a single destination.*
 
 
-## Old issue closing policy
+## Issue closing policy
 
-We have yet to settle on a precise guide for closing out old issues, but, for
-now, if you happen across something that could easily be marked as resolved,
-please do so.
+We need to strike a balance in issue handling between keeping a maintainable
+set of issues&mdash;thereby making the lives of maintainers, contributors, and
+users easier&mdash;and allowing legitimate issues to remain open.
+
+The policy for closing an issue depends entirely on the type of issue we're
+looking at.
+
+When closing an issue, please provide a brief explanation for why the issue was
+closed, such as a reference to a PR, a comment about lack of clarification,
+etc.
+
+### Discussion
+
+Some issues are for discussing the project. These issues should start with a
+clear question, have an owner, and be closed when either:
+
+* The original question has been answered
+* There is no clear question or owner
+* No progress has been made in 30 days
+
+Discussions are typically opened by maintainers and contributors.
+
+### Support
+
+Support questions are requests for assistance in using Stack. They must have a
+clear set of steps followed by the user and a desired outcome. Support requests
+sometimes overlap with bug reports, handled next. Supports requests are closed
+when:
+
+* The user decides their question is answered
+* No clarification to questions from the Stack team is given for 7 days
+* The question is determined out of scope by the Stack team
+
+### Bug report
+
+A bug report describes a misbehavior in Stack itself. There are some cases
+where an issue may appear somewhere between a support request and a bug report.
+The Stack team will determine which category an issue falls into. Bug reports
+must include clear reproduction steps, an expected outcome, and an actual
+outcome. Bug reports are closed when:
+
+* The bug is fixed on the `master` or `stable` branch (NOTE: we do not wait for
+  fixes to be released before closing issues)
+* The Stack team is unable to reproduce the issue
+* No clarification to questions from the Stack team is given for 7 days
+* The bug is determined to be out of scope by the Stack team, such as being an
+  upstream issue
+
+### Feature request
```

Comment:
I like this idea. The requests are all listed together, and can be prioritised, grouped and thinned every so often as appropriate. This would help with directing people who want to contribute to a feature request to implement, too.

However, I'd be tempted to move the wishlist into the `stable` branch (or a new branch) and have feature requests as PRs to that (in a similar way to how ghc-proposals works). The PR can be merged if the feature request is accepted. Then another PR can handle the implementation. This would

1. give maintainers a way to curate PRs (I'm not sure how well the wiki does this),
2. ensure that the wishlist is distributed as part of the project (which it is), and
3. build the wishlist into the documentation (meaning there may be some feedback from users, too!).

Finally, the three options for a PR are close, merge and leave open. With PRs to a feature list rather than as issues, the default shifts from 'open' to 'merge.

Reply:
I'll give the counter argument, and reason I went this way in the first place: requiring a PR to add a feature to the wishlist adds burden back on the maintainer team, and prevents a "self service" approach to adding features. I see the arguments you give though, and typically prefer docs in a repo versus a proper Wiki. So I'm pretty much on the fence here.

Reply:
You've hit on the exact point where I disagree. For me, maintaining a list of features ought to be the responsibility of the maintainer team — as it is our responsibility to maintain the features themselves.

However, both are an big improvement on the current situation, and so I'd accept whichever.

Reply:
OK, let's go with the Wiki for now. I'll add a comment to the top of the Wiki that we may elect to move the wishlist in-repo at some point in the future if the Wiki is too unmaintainable.